s3/proto/s3.proto

```diff
@@ -315,16 +315,53 @@ message BaseRequest {
 }
 
 message ListObjectsRequest {
-    // Bucket is a required field
-    string bucket = 1;
-    string delimiter = 2;
-	map<string, string> filter = 3;
-	int32  offset = 4;
-	int32  limit = 5;
+	string bucket = 1;
+	bool versioned = 2; // should return versioned objects?
+	int32 version = 3;  // Currently 1 or 2
+	string delimiter = 4;
+	string encodingType = 5;
+	int32 maxKeys = 6;
+	string prefix = 7;
+
+	// v1 specific
+	string marker = 8;
+
+	// v2 specific
+	string continuationToken = 9;
+	string startAfter = 10;
+	bool fetchOwner = 11;
+
+	// versioned specific
+	string keyMarker = 12;
+	string versionIdMarker = 13;
+
+    // for internal use

```

Comment:
please update comment with details...used for ?

Reply:
Done